modules/service/src/main/scala/db/migration/R__SmartGmosNorth.scala

```diff
@@ -0,0 +1,109 @@
+// Copyright (c) 2016-2023 Association of Universities for Research in Astronomy, Inc. (AURA)
+// For license information see LICENSE or https://opensource.org/licenses/BSD-3-Clause
+
+package db.migration
+
+import cats.data.NonEmptyList
+import cats.effect.IO
+import cats.effect.Resource
+import eu.timepit.refined.types.numeric.PosLong
+import fs2.Pipe
+import lucuma.core.enums.Instrument.GmosNorth
+import lucuma.odb.smartgcal.FileReader
+import lucuma.odb.smartgcal.data.GmosNorth.FileEntry
+import lucuma.odb.smartgcal.data.GmosNorth.TableKey
+import lucuma.odb.smartgcal.data.GmosNorth.TableRow
+import lucuma.odb.smartgcal.data.SmartGcalValue
+import lucuma.odb.util.Codecs.*
+import lucuma.odb.util.GmosCodecs.*
+import skunk.Encoder
+import skunk.codec.temporal.interval
+import skunk.implicits.*
+
+import java.io.InputStream
+
+/**
+ * Repeatable Smart GCal configuration loader for GMOS North.  Located,
+ * instantiated and executed by flyway.
+ */
+class R__SmartGmosNorth extends SmartGcalMigration[TableRow]("GMOS North") {
+
+  val definitionFiles: NonEmptyList[(String, IO[InputStream])] =
+    gcalFilesFromClasspath("GMOS-N_ARC", "GMOS-N_FLAT")
```

Comment:
These are taken from the classpath at the moment, but we could consider reading them from, say, AWS at some point and allow them to be uploaded, checked and applied like they are in `ocs`.

Reply:
Yeah maybe they could be in another GH repo. The same commit hash logic would work but we would have some insulation in case they want to roll back a change or whatever. What you have now is fine of course.